src/webrtc/callFeed.ts

```diff
@@ -76,20 +64,28 @@ type EventHandlerMap = {
     [CallFeedEvent.Disposed]: () => void;
 };
 
-export class CallFeed extends TypedEventEmitter<CallFeedEvent, EventHandlerMap> {
-    public feedId: string;
-    public readonly userId: string;
-    public readonly deviceId: string | undefined;
-    public purpose: SDPStreamMetadataPurpose;
+export abstract class CallFeed extends TypedEventEmitter<CallFeedEvent, EventHandlerMap> {
+    public abstract readonly id?: string;
+    public abstract readonly streamId?: string;
+    public abstract readonly purpose: SDPStreamMetadataPurpose;
+    public abstract readonly connected: boolean;
+    public abstract readonly userId: string;
+    public abstract readonly deviceId?: string;
+
+    public abstract isLocal: boolean;
+    public abstract isRemote: boolean;
+
     public speakingVolumeSamples: number[];
-    public tracksMetadata: SDPStreamMetadataTracks = {};
-
-    private _stream?: MediaStream;
-    private client: MatrixClient;
-    private call?: MatrixCall;
-    private roomId?: string;
-    private audioMuted: boolean;
-    private videoMuted: boolean;
+
+    protected readonly _id: string;
+    protected _tracks: CallTrack[] = [];
+    protected _stream?: MediaStream;
```

Comment:
One suggestion: we should be consistent with underscores. In my opinion, underscores are not necessary in typescript because we have operators like `private`, `public readonly`, `public`.

I don't mind using underscores to indicate that it's a very private property either. Only there doesn't seem to be a rule in `matrix-js-sdk` to use them or not.

Reply:
I think we're generally against using underscores for private props, the only case where we use them is when we have a getter and a private prop of the same name, which is the case here

Reply:
OK, I see 👍